src/providers/hana/qgshanacolumntypethread.h

```diff
@@ -0,0 +1,55 @@
+/***************************************************************************
+   qgshanacolumntypethread.h
+   --------------------------------------
+   Date      : 31-05-2019
+   Copyright : (C) SAP SE
+   Author    : Maxim Rylov
+ ***************************************************************************/
+
+/***************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ ***************************************************************************/
+#ifndef QGSHANACOLUMNTYPETHREAD_H
+#define QGSHANACOLUMNTYPETHREAD_H
+
+#include "qgsdatasourceuri.h"
+#include "qgshanatablemodel.h"
+#include <QThread>
+
+// A class that determines the geometry type of a given database
+// schema.table.column, with the option of doing so in a separate
+// thread.
+class QgsHanaColumnTypeThread : public QThread
+{
+    Q_OBJECT
+  public:
+    QgsHanaColumnTypeThread( const QString &connName, const QgsDataSourceUri &uri, bool allowGeometrylessTables, bool userTablesOnly );
+
+    // These functions get the layer types and pass that information out
+    // by emitting the setLayerType() signal.
+    void run() override;
+
+  signals:
+    void setLayerType( QgsHanaLayerProperty layerProperty );
+    void progress( int, int );
+    void progressMessage( const QString & );
+
+  public slots:
+    void stop();
```

Comment:
This should be reworked.

The Qt documentation states that classes derived from QThread should not introduce new slots. The method seems to be called directly anyway without using a Qt signal. As this function is called from a different thread, some synchronization is required as well.

However, we don't need it at all, because QThread already comes with the necessary methods:
- `requestInterruption()` to ask the thread to stop
- `isInterruptionRequested()` to check in the run method if the thread should stop

Be careful before destroying an instance. If it is still running, calling `requestInterruption()` is not sufficient (or calling the current `stop()`). You have to call `wait()` afterwards. I saw that in the destructor of `QgsHanaSourceSelect`. We would end in a qfatal if the thread was still running.

Reply:
Done in 5a273e3.